### PR TITLE
Removed sending data to the server in case of load

### PR DESCRIPTION
**Pull request by hamadodene**
Removed sending data to the server in case of load.
In case of load, on the server side, the data is not necessary as it is never used.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

### CODE REVIEW

**Comment by eolivelli** [comment]
I kindly suggest to add test cases to cover this changes and the protocol.
We only have end-to-end tests, and it will be super easy to break the protocol and the behaviour without tests

**Comment by hamadodene** [comment]
@eolivelli  @dmercuriali  I added a new class RequestParametersTest where I text the message format of requests of type PUT, LOAD, INVALID AND FETCH

File: blazingcache-core/src/test/java/blazingcache/client/RequestParametersTest.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+            cacheServer.start();
+            try (CacheClient client = new CacheClient("theClient1", "ciao", new NettyCacheServerLocator(serverHostData));) {
+                client.start();
+                assertTrue(client.waitForConnection(10000));
+                assertNull(client.get("foo"));
+
+                client.setInternalClientListener(new InternalClientListener() {
+                    @Override
+                    public void onRequestSent(Message message) {
+                        assertTrue(message.type == Message.TYPE_PUT_ENTRY);
```
Comment (eolivelli) [inline blazingcache-core/src/test/java/blazingcache/client/RequestParametersTest.java:53]: nit: assertEquals ?

File: blazingcache-core/src/test/java/blazingcache/client/RequestParametersTest.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                        assertTrue(message.parameters.get("data") != null);
+                        assertTrue(message.parameters.get("key") != null);
+                        assertTrue(message.parameters.get("expiretime") != null);
```
Comment (eolivelli) [inline blazingcache-core/src/test/java/blazingcache/client/RequestParametersTest.java:56]: we should also validate the number of entries. To test that there is nothing more that what is expected

File: blazingcache-core/src/test/java/blazingcache/client/RequestParametersTest.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                client.start();
+                assertTrue(client.waitForConnection(10000));
+                assertNull(client.get("foo"));
+
+                client.setInternalClientListener(new InternalClientListener() {
+                    @Override
+                    public void onRequestSent(Message message) {
+                        assertTrue(message.type == Message.TYPE_PUT_ENTRY);
+                        assertTrue(message.parameters.get("data") != null);
+                        assertTrue(message.parameters.get("key") != null);
```
Comment (eolivelli) [inline blazingcache-core/src/test/java/blazingcache/client/RequestParametersTest.java:55]: we know the key, we can use assertEquals

File: blazingcache-core/src/test/java/blazingcache/client/RequestParametersTest.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+            try (CacheClient client = new CacheClient("theClient1", "ciao", new NettyCacheServerLocator(serverHostData));) {
+                client.start();
+                assertTrue(client.waitForConnection(10000));
+                assertNull(client.get("foo"));
+
+                client.setInternalClientListener(new InternalClientListener() {
+                    @Override
+                    public void onRequestSent(Message message) {
+                        assertTrue(message.type == Message.TYPE_PUT_ENTRY);
+                        assertTrue(message.parameters.get("data") != null);
```
Comment (eolivelli) [inline blazingcache-core/src/test/java/blazingcache/client/RequestParametersTest.java:54]: we know the payload, we can use assertEquals

File: blazingcache-core/src/test/java/blazingcache/client/RequestParametersTest.java (hunk 2 above)
Comment (eolivelli) [reply blazingcache-core/src/test/java/blazingcache/client/RequestParametersTest.java:56]: for expiretime we can at least verify the type

**Comment by hamadodene** [comment]
@dmercuriali  @eolivelli PTAL

**Pull request merged**